syllabus/syllabus-hbm/src/java/org/sakaiproject/component/app/syllabus/SyllabusAttachmentImpl.java

```diff
@@ -27,7 +27,7 @@
 import org.sakaiproject.api.app.syllabus.SyllabusData;
 
 @Data
-@EqualsAndHashCode(of = "syllabusAttachId")
+@EqualsAndHashCode(of = {"syllabusAttachId", "name"})
```

Comment:
attachmentId is a better candidate as name can be easily duplicated

Reply:
the combination of the two is meant to be unique. and the site import code does not have an attachmentId so it will be null just like syllabusAttachId

Reply:
yes the site import code would need to create the attachment first (just like resources), there is map that tracks these kinds of things so that the new id can be fetched.

Reply:
probably assignments would have an example as it would have to solve the same issue